pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -33,9 +33,17 @@ def on_trainer_init(
         check_val_every_n_epoch: int,
         reload_dataloaders_every_n_epochs: int,
         reload_dataloaders_every_epoch: bool,
-        prepare_data_per_node: bool,
+        prepare_data_per_node: Optional[bool] = None,
     ) -> None:
         self.trainer.datamodule = None
+
+        if prepare_data_per_node is None:
+            prepare_data_per_node = True
+        else:
+            rank_zero_deprecation(
+                "Setting `prepare_data_per_node` with the trainer flag is deprecated and will be removed in v1.7.0! "
+                "Please set `prepare_data_per_node` in LightningDataModule directly instead. "
```

Comment:
```suggestion
                "Please set `prepare_data_per_node` in LightningDataModule or LightningModule directly instead. "
```
as mentioned, DataHooks are available on both LightningModule and LightningDataModule